rocks/RocksStorage.java

```diff
@@ -181,9 +181,11 @@ public Cache(OptimisticTransactionDB rocksDB) {
 
         @Override
         public <G> ResourceIterator<G> iterate(byte[] key, BiFunction<byte[], byte[], G> constructor) {
-            assert isOpen();
+//            assert isOpen();
+            if (!isOpen()) throw GraknException.of(TRANSACTION_CLOSED);
             RocksIterator<G> iterator = new RocksIterator<>(this, key, constructor);
             iterators.add(iterator);
+            if (!isOpen()) throw GraknException.of(TRANSACTION_CLOSED); //guard against close() race conditions
```

Comment:
Another possible solution is to make `iterate` `synchronized`, however this is deemed unacceptable for performance reasons.